### ISSUE TITLE
Don't request permission to access all web sites (Firefox)

### ISSUE THREAD

**Issue by marco-c** [sourcegraph/sourcegraph#760]

**Comment by ijsnow** [comment]
Hello! This is still an issue that's blocked by https://bugzilla.mozilla.org/show_bug.cgi?id=1432083, but I just wanted to post an update on the Firefox extension. 

We decided to move to self hosting the Firefox extension. In order to get the latest version and future updates, you'll have to uninstall your current version and install the new one by clicking the link on [the integrations page](https://docs.sourcegraph.com/integration/browser_extension) from your Firefox browser.

**Comment by SeriousM** [comment]
This is the one from Sourcegraph:
![image](https://user-images.githubusercontent.com/462013/55146453-e8fc8500-5144-11e9-8b4d-6c66dbd9d7e2.png)

And here the one from Redefined GitHub:
![image](https://user-images.githubusercontent.com/462013/55146481-f44fb080-5144-11e9-9b19-8a06a644cc79.png)

How is this so much different that you need ALL websites?

**Comment by ijsnow** [comment]
Duplicate of https://github.com/sourcegraph/sourcegraph/issues/760

**Comment by SeriousM** [comment]
@ijsnow you marked this issue as duplicate of itself :)

**Comment by SeriousM** [comment]
Btw, Redefined GitHub can allow the user to enable the extension on another domain as the initial one. Maybe this helps you?

![image](https://user-images.githubusercontent.com/462013/55172632-a2298200-517a-11e9-8349-0af0688e7a76.png)

**Comment by sqs** [comment]
Thank you, @SeriousM.

Reassigned to the current code owners @lguychard and @felixfbecker to look into it. This comes up regularly as a question from Firefox users in our extension uninstall feedback form, and in general it is quite likely that the permissions grant is a big barrier to installing it (I know I very rarely install extensions that ask for all perms).

**Comment by SeriousM** [comment]
Thank you @sqs .
Thats the very reason why I didn't installed it and added this information.
I guess no one will install an extension that asks for everything because it could be hijacked later and sends the browsing data to third parties or injects ads.
Not that your extension is doing it or you ever want to, but no one knows what the future holds.

**Comment by fregante** [comment]
I see you don't have broad permissions anymore, is this issue solved? 

https://github.com/sourcegraph/sourcegraph/blob/c9d3a78438d918b5a954ca72622a33fa350da4f7/browser/src/extension/manifest.spec.json#L68-L83

**Comment by felixfbecker** [comment]
Hi @fregante, that is the base manifest, but for Firefox we still request permissions for all pages: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@4d84cd2b62888f584c514e67e46119dff43a3049/-/blob/browser/scripts/tasks.ts#L84

The [bug linked earlier in the thread](https://bugzilla.mozilla.org/show_bug.cgi?id=1382953) required workarounds that made onboarding harder. All I can find from previous code:

> There is an outstanding bug with optional permissions on Firefox that breaks requesting additional permissions from a popup or the preferences menu. The only workaround right now is opening a new tab for the options menu.

That issue is fixed so this can be revisited now. There's an open PR https://github.com/sourcegraph/sourcegraph/pull/4760 but it wasn't tested yet.

**Comment by fregante** [comment]
The new version of my module works in Firefox, it fixed compatibility a while ago

**Comment by SeriousM** [comment]
What's "your module"?

**Comment by fregante** [comment]
The one I mentioned earlier, SourceGraph is already using it https://github.com/fregante/webext-domain-permission-toggle

**Comment by eight04** [comment]
Why not just verify `location.href` before running content scripts?

**Comment by fregante** [comment]
> Why not just verify `location.href` before running content scripts?

That already happens. This issue is about dropping a permission so that check isn't necessary.

@SeriousM this issue would be solved if you dropped the UI to add domains and just used https://github.com/fregante/webext-domain-permission-toggle, it's not affected by https://bugzilla.mozilla.org/show_bug.cgi?id=1432083

**Comment by eight04** [comment]
If the extension already performs a URL check, then I think #8489 is not a duplicate of this issue.

**Comment by fregante** [comment]
Solving this issue would also solve yours, but yeah if this takes longer, your issue could be solved alternatively.

**Comment by fregante** [comment]
> The [bug linked earlier in the thread](https://bugzilla.mozilla.org/show_bug.cgi?id=1382953) required workarounds that made onboarding harder. All I can find from previous code:

This bug appears to have been fixed in Firefox 75, so the current workarounds can be dropped and Firefox’ permissions can match Chrome’s.
